rfcs/react-components/convergence/semantic-shared-colors.md

```diff
@@ -0,0 +1,195 @@
+| `yellow`     | warning               | Replace by semantic `warning`, mapping changed from `yellow` to `orange`                                                            |
+| `berry`      | OOF                   | No change                                                                                                                           |
+| `lightGreen` | available             | No change                                                                                                                           |
+| `marigold`   | away, persona         | No change                                                                                                                           |
+
+Note: even though the task now is to introduce the semantic names only for the **status** tokens, in the future we might be asked to do the same change for **presence** tokens.
+
+### Other platforms
+
+I briefly checked react native packages for iOS and Android produced by token pipeline, they use tokens like `warningBackground1`.
+
+## Detailed Design or Proposal
+
+There are multiple options how to address the problem. This chapter describes the options and their pros and cons.
+
+### 1. Introduce new semantic status tokens
+
+Add 3 new semantic colors:
+
+1. danger
+2. success
+3. warning
+
+For each of the 3 new colors, add 9 (or 10) color slots (see the table above).
+
+Update the sources in the library to use the semantic color names instead of the current shared color names.
+
+Find a way how to get rid of the legacy tokens which are no longer needed:
+
+- `green` (10 tokens)
+- `yellow` (10 tokens)
+
+Once the legacy tokens are not used by the library code, there are three possible ways how to get rid of the tokens:
+
+#### 1A: "deprecate" the tokens and remove them in the next major release
+
+#### 1B: "deprecate" the tokens and create a V2 theme which does not contain them - let applications switch when they are ready
+
+#### 1C: big bang - break semver, remove the tokens immediately
+
+#### Pros and Cons
+
+- 👍 Decouples the persona and status shared colors, moves us to architecturally cleaner solution.
+- 👍 Designers can "freely" change the status colors' mapping.
+- 👍 Partners can override persona and status colors independently.
+- 👎 We will remove some "basic" colors `green`, `yellow` - are we sure those have been used as semantic colors only?
+  approach
+- 👎 1A: adds ~30 new tokens
+- 👎 If design decides to do the same for **presence** tokens, we will need to add another ~40 tokens to follow the same
+- 👎 1A: can cause temporarily inconsistent UI colors between library and partners
+- 👍 1B: we can use the current mapping in v1 theme and the new mapping in v2 theme - partners can control the moment when they switch to the new mapping
+- 👎 1B: requires us maintaining the double amount of themes
+- 👎 1C: breaks partners (the questions is how much, but...)
```

Comment:
can geoff's query tool track down uses of those tokens?

Reply:
I checked just TMP - there are 12 usages of green, 5 usages of yellow, but 121 usages of red.
So 1C would be difficult.

But even without breaking changes, we need to be careful with switching from `red` to `cranberry` regardless the name (danger or not) - some usages mean danger, other usages might mean persona or status - partners need to review all the usages and decide what the intent is and change accordingly.
To avoid inconsistent UI, best way forward would be to introduce `danger` and map it to `red`, let partners update their usages and only after that change `danger` from red to cranberry.